glean-core/src/metrics/time_unit.rs

```diff
@@ -0,0 +1,58 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+use crate::error::{Error, ErrorKind};
+
+/// Enumeration of different resolutions supported by the
+/// time related metric types (e.g. DatetimeMetric).
+//#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum TimeUnit {
+    Nanosecond,
+    Microsecond,
+    Millisecond,
+    Second,
+    Minute,
+    Hour,
+    Day,
+}
+
+impl TimeUnit {
+    /// How to format the given TimeUnit, truncating
+    /// the time if needed.
+    pub fn format_pattern(&self) -> &'static str {
+        use TimeUnit::*;
+        match self {
+            Nanosecond => "%Y-%m-%dT%H:%M:%S%.f%:z",
+            Microsecond => "%Y-%m-%dT%H:%M:%S%.6f%:z",
+            Millisecond => "%Y-%m-%dT%H:%M:%S%.3f%:z",
+            Second => "%Y-%m-%dT%H:%M:%S%:z",
+            Minute => "%Y-%m-%dT%H:%M%:z",
+            Hour => "%Y-%m-%dT%H%:z",
+            Day => "%Y-%m-%d%:z",
+        }
+    }
+}
+
+/// Trait implementation for converting an integer value
+/// to a TimeUnit. This is used in the FFI code.
```

Comment:
It's rare that trait implementations get a comment in Rust, but I guess it doesn't hurt.

Should there be a note that the values should match the ordering on the kotlin side?